frontend/src/test/scala/bloop/bsp/BspIntellijClientSpec.scala

```diff
@@ -0,0 +1,63 @@
+package bloop.bsp
+
+import java.io.File
+import java.net.URLClassLoader
+import java.nio.file.Files
+import java.nio.file.StandardOpenOption
+
+import bloop.cli.BspProtocol
+import bloop.data.WorkspaceSettings
+import bloop.io.AbsolutePath
+import bloop.logging.BspClientLogger
+import bloop.logging.RecordingLogger
+import bloop.util.{TestProject, TestUtil}
+import ch.epfl.scala.bsp.WorkspaceBuildTargetsResult
+
+object LocalBspIntellijClientSpec extends BspIntellijClientSpec(BspProtocol.Local)
+object TcpBspIntellijClientSpec extends BspIntellijClientSpec(BspProtocol.Tcp)
+
+class BspIntellijClientSpec(
+    override val protocol: BspProtocol
+) extends BspBaseSuite {
+
+  test("refresh project data on buildTargets request") {
+    TestUtil.withinWorkspace { workspace =>
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val `A` = TestProject(workspace, "a", Nil)
+      val `B` = TestProject(workspace, "b", Nil)
+      val initialProjects = List(`A`)
+      val configDir = TestProject.populateWorkspace(workspace, initialProjects)
+
+      val refreshProjectsCommand = {
+        val content = sanitizeForBash(`B`.toJson)
```

Comment:
I think we can avoid the `sanitizeForBash` method by writing the contents to a temporary file and then `cp` that file over. What do you think?

Reply:
Yeah, the sanitizing might not be reliable. I updated the PR with `mv` on tmp file.